src/coreclr/jit/codegenxarch.cpp

```diff
@@ -3040,42 +3045,67 @@ void CodeGen::genCodeForCpBlkUnroll(GenTreeBlk* node)
             }
         }
 
-        // TODO-CQ-XArch: On x86 we could copy 8 byte at once by using MOVQ instead of four 4 byte MOV stores.
-        // On x64 it may also be worth copying a 4/8 byte remainder using MOVD/MOVQ, that avoids the need to
-        // allocate a GPR just for the remainder.
-    }
-
-    if (size > 0)
-    {
-        regNumber tempReg = node->GetSingleTempReg(RBM_ALLINT);
-
-        for (unsigned regSize = REGSIZE_BYTES; size > 0; size -= regSize, srcOffset += regSize, dstOffset += regSize)
+        if (size > 0)
         {
-            while (regSize > size)
-            {
-                regSize /= 2;
-            }
+            // Copy the remainder by moving the last regSize bytes of the buffer
+            unsigned remainder = regSize - size;
+            assert(remainder <= regSize);
+
+            srcOffset -= remainder;
```

Comment:
is the idea that after it `srcOffset + regSize == totalSize`? 
if we have blockSize = 48, would it be beneficial to copy as [0, 32) YMM  [32, 48) XMM, not as [0,32) YMM [16, 48) YMM?

Reply:
> Could you please check if it fixes #33617?

Sure, I'll take a look and add a separate comment after I check.

> is the idea that after it `srcOffset + regSize == totalSize`?

Yes, that's the idea. If there are any remaining bytes leftover after the main copy loop, back up the offset enough so that we can do a single SIMD move with whatever register was allocated to finish removing the remaining bytes.

> would it be beneficial to copy as [0, 32) YMM [32, 48) XMM, not as [0,32) YMM [16, 48) YMM?

I don't expect this to have a perf difference, although I'd have to benchmark it to be sure. It seemed easier/cleaner to just reuse the register and regSize that was setup pre-copy. Let me know if you think it's better to break up cases like this instead.